### ISSUE TITLE
Array.ctypes.data is ususally called without checking data types

### ISSUE THREAD

**Issue by johannesmayer** [SyneRBI/SIRF#1006]
Passing data from Python to C++ using the `array.ctypes.data` method is often used without the check of the correct datatype.
Examples are the constructor of `AffineTransformation` where when constructed as

```
translation = [10, 20, 30]
euler = [12, 15,10]
aff = AffineTransformation( translation, euler)
```
yields an identity transformation since no check is made that `translation` and `euler` are of type `numpy.float32` (which they are not).

This is addressed in #1005 where a Python utility method is provided that can be called by the class methods to ensure proper data format and array contiguousness and applied to the AffineTransformation.

However, a method of this type should be used whenever the `ctypes.data` is invoked, depending on what data pointer the receiving glue method asks for.

**Comment by evgueni-ovtchinnikov** [comment]
I have looked through all usages of `ctypes.data`, and so far as I can see the only places requiring checks are those I pointed at in my comment to #1005. I believe this issue can be closed after #1005 is dealt with and merged.

**Comment by KrisThielemans** [comment]
we'll merge #1005 and leave https://github.com/SyneRBI/SIRF/pull/1005#issuecomment-909082133 for you @evgueni-ovtchinnikov  😄
